hardware/data_acquisition_units.py

```diff
@@ -19,7 +20,28 @@
 from ctypes import byref
 import queue
 import time
+from hardware import u
+import random
 
+class MockDAQ:

```

Comment:
One thing that would be really handy for me is if we could make this Mock DAQ return several different types of data.

- white noise
- random walk noise
- flicker noise

It would be grand if it could return data synchronously, or do so asynchronously, as we define below for the real class.

This doesn't have to happen for this pull request, but we should at least put it into an issue tracker

Reply:
To add onto this, data_acquisition_units.py ideally wouldn't even know about the MockDAQ class. It should be in an isolated module that import whatever DAQ it wants to mock. Doesn't necessarily need to be addressed in this PR.